skbio/stats/distance/_regression.py

```diff
@@ -0,0 +1,170 @@
+        in all of the distance matrices. If ``False``, any nonmatching IDs are
+        discarded before running the test. See `n` (in Returns section below)
+        for the number of matching IDs that were used in the test. This
+        parameter is ignored if `x` and `y` are ``array_like``.
+    lookup : dict, optional
+        Maps each ID in the distance matrices to a new ID. Used to match up IDs
+        across distance matrices prior to running the Mantel test. If the IDs
+        already match between the distance matrices, this parameter is not
+        necessary. This parameter is disallowed if `x` and `y` are
+        ``array_like``.
+
+    Returns
+    -------
+    B : np.array
+        Array of coefficients
+    T : np.array
+        Array of pseudo t-statistics for each coefficient
+    pvals : np.array
+        Array of p-values for each coefficient calculated
+        using pseudo T-tests
+    F : float
+        pseudo F-statistic for lack of fit
+    model_pval : float
+        pvalue from pseudo F-test for lack of fit
+    R2: float
+        Coefficient of determination squared
+
+    See Also
+    --------
+    DistanceMatrix
+
+    References
+    ----------
+    .. [1] Legendre, P. and Legendre, L. (2012) Numerical Ecology. 3rd English
+       Edition. Elsevier.
+    .. [2] Legendre, P. Lapointe, F., Casgrain P. (1994) Modeling Brain
+       Evolution from Behavior: A Permutational Regression Approach
+    .. [3] https://cran.r-project.org/web/packages/ecodist/index.html
+    """
+
+    # Unpack kwargs
+    params = {'permutations': 10000,
+              'random_state': 0,
+              'strict': True,
+              'lookup': {}}
+    for key in ('permutations', 'random_state',
+                'strict', 'lookup'):
+        params[key] = kwargs.get(key, params[key])
+    permutations = params['permutations']
+    random_state = params['random_state']
+    strict = params['strict']
+    lookup = params['lookup']
+
+    random_state = check_random_state(random_state)
+
+    # Read all of the distance matrices from files
+    # if necessary
+    if isinstance(y, six.string_types):
+        y = DistanceMatrix.read(y)
+    xargs = []
+    for x in args:
+        if isinstance(x, six.string_types):
+            x = DistanceMatrix.read(x)
+        xargs.append(x)
+
+    # Conform all of the ids in the distance matrices to the same order
+    if strict:
+        for i in range(len(xargs)):
+            y, xargs[i] = _order_dms(y, xargs[i],
+                                     strict=strict,
+                                     lookup=lookup)
+
+    # Linearize all predictor distance matrices into
+    # a single matrix
+    n = len(y.data)
+    X = np.vstack([np.ones((1, n*(n-1)/2))] + \
+                  [k.data[np.triu_indices(n, 1)] for k in xargs]).T
+    Y = np.atleast_2d(y[np.triu_indices(n, 1)]).T
+    n, p = X.shape
+    J = np.ones((n, n))
+    I = np.identity(n)
+
+    # Define regression function
```

Comment:
:+1: That'll actually clean things up. And also open the doors for permutative OLS.

The only issue that I can see is that having a permutative OLS would no longer fit within the `skbio.stats.distance` package.  It'll have to become its own separate thing...